### PR TITLE
feat(Hyakanime.fr): Add presence

### LINKED ISSUE #6193: Hyakanime | hyakanime.fr
### Website name

Hyakanime

### Website URL

http://hyakanime.fr/

### Website logo

https://i.imgur.com/cRrhUIZ.png

### Prerequisites

- [ ] It is a paid service
- [ ] It displays NSFW content
- [ ] It is region restricted

### Description

Hyakanime is a french anime list website made for fans by fans.

First of all, an user started developing a presence for this site but we're not actually sure if it's possible to publish it in the store like the one requested from here. So we're making a request but if the dev can make his presence by himself then we would like to know how can he pull the presence to the store.
You can contact the dev on discord : https://discord.gg/Dn4kWhGBmX

If it's not possible to use our dev's version and need to make a new presence with this request :
We would like :
-Anime page with something like "Viewing an anime page - [anime name]"
-User page with something like "Viewing user [user]"
-Home page
-General browsing/searching

Here is a folder with all of the logos if you need them : https://imgur.com/a/rsCbR9G
Thank You

### PR DESCRIPTION
**Pull request by dem0nx309**
**Describe the changes in this pull request:**

I have added a rich prensence for hyakanime.fr

<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<br>

 Capture d’écran 2022-05-11 142529 

Capture d’écran 2022-05-11 142613

Capture d’écran 2022-05-11 142636

Capture d’écran 2022-05-11 142653

Capture d’écran 2022-05-11 142715

Capture d’écran 2022-05-11 142735

Capture d’écran 2022-05-11 142759

Capture d’écran 2022-05-11 142811
 
</details>

### CODE REVIEW

File: websites/H/Hyakanime/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"name": "Dem0nx309",
+		"id": "145559039173066752"
+	},
+	"service": "Hyakanime",
+	"description": {
+		"en": "French site, allows you to follow your anime progress",
+		"fr": "Site franÃ§ais, permet de suivre ta progresion anime"
```
Comment (ImRodry) [inline websites/H/Hyakanime/dist/metadata.json:10]: Taken from the website's embed, you should either use "Crée ta bibliothèque d'animés !" or "Bibliothèque et Agenda Anime" (and their corresponding translation). Also I'm pretty sure that you didn't mean to spell "franÃ§ais"

File: websites/H/Hyakanime/presence.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+			}
+		];
+	} else if (page.startsWith("/profil/")) {
+		presenceData.details = "Regarde le profil de:";
+		presenceData.state = document.querySelector(
+			"#root > div:nth-child(3) > div.profil-banner-top > div > div.bloc-name-banner > h1"
+		).textContent;
+		presenceData.buttons = [
+			{
+				label: "Profil",
```
Comment (ImRodry) [inline websites/H/Hyakanime/presence.ts:30]: Button labels should have a verb in the infinitive e.g. "View profile" (translate accordingly)

File: websites/H/Hyakanime/presence.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+			{
+				label: "Agenda",
+				url: document.location.href
+			}
+		];
+	}
+
+	if (presenceData.details) presence.setActivity(presenceData);
+	else {
+		presenceData.details = "Sur la page d'accueil";
```
Comment (ImRodry) [inline websites/H/Hyakanime/presence.ts:53]: this should be set in the if chain above in the form of an else at the end

**Review by ImRodry — CHANGES_REQUESTED** [review]

File: websites/H/Hyakanime/presence.ts (hunk 3 above)
Comment (dem0nx309) [reply websites/H/Hyakanime/presence.ts:55]: I am not sure I understand, would it be possible to explain it to me again please

**Review by ImRodry — CHANGES_REQUESTED** [review]
From taking a quick look at the website it seems like you're not supporting all pages, mainly some at the bottom of the website and ones like /diffusion/*. Please try your best to support all pages on the website

**Review by itsmelouis — COMMENTED** [review]
Translations look good to me 👍

File: websites/H/Hyakanime/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"name": "Dem0nx309",
+		"id": "145559039173066752"
+	},
+	"service": "Hyakanime",
+	"description": {
+		"en": "French website, allows you to create your own library of animes",
+		"fr": "Site français, permet de créer ta bibliothèque d'animés"
```
[the remark this answers is not in the document]
Comment (Rudy2612) [reply websites/H/Hyakanime/dist/metadata.json]: Yes i think

File: websites/H/Hyakanime/presence.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "applogo_7fac0ec4359bda8ccf0f",
+			startTimestamp: browsingTimestamp
+		},
+		page = document.location.pathname;
+	presenceData.details = "Sur la page d'accueil";
```
Comment (Slowlife01) [inline websites/H/Hyakanime/presence.ts:12]: Please move `details` to the object

**Review by ImRodry — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**